src/meter/__examples__/__tests__/Meter.test.tsx

```diff
@@ -0,0 +1,471 @@
+    expect(meter).toHaveAttribute("aria-valuemax", "1");
+    expect(meter).toHaveAttribute("aria-valuenow", "0.5");
+    expect(meter).toHaveAttribute("aria-valuetext", "50%");
+  });
+
+  it("value below min", function () {
+    const { getByRole } = render(<MeterComp value={-0.5} />);
+    const meter = getByRole("meter");
+
+    expect(meter).toHaveAttribute("aria-valuemin", "0");
+    expect(meter).toHaveAttribute("aria-valuemax", "1");
+    expect(meter).toHaveAttribute("aria-valuenow", "0");
+    expect(meter).toHaveAttribute("aria-valuetext", "0%");
+  });
+
+  it("value above max", function () {
+    const { getByRole } = render(<MeterComp value={1.5} />);
+    const meter = getByRole("meter");
+
+    expect(meter).toHaveAttribute("aria-valuemin", "0");
+    expect(meter).toHaveAttribute("aria-valuemax", "1");
+    expect(meter).toHaveAttribute("aria-valuenow", "1");
+    expect(meter).toHaveAttribute("aria-valuetext", "100%");
+  });
+
+  it("custom min and max", function () {
+    const { getByRole } = render(<MeterComp min={0} value={5} max={10} />);
+    const meter = getByRole("meter");
+
+    expect(meter).toHaveAttribute("aria-valuemin", "0");
+    expect(meter).toHaveAttribute("aria-valuemax", "10");
+    expect(meter).toHaveAttribute("aria-valuenow", "5");
+    expect(meter).toHaveAttribute("aria-valuetext", "50%");
+  });
+
+  it("supports aria-label", function () {
+    const { getByRole } = render(<MeterComp aria-label="Meter" />);
+    const meter = getByRole("meter");
+
+    expect(meter).toHaveAttribute("aria-label", "Meter");
+  });
+
+  it("supports custom DOM props", function () {
+    const { getByTestId } = render(<MeterComp data-testid="test" />);
+    const meter = getByTestId("test");
+
+    expect(meter).toBeInTheDocument();
+  });
+
+  it("meter state initial state", function () {
+    const result = renderMeterStateHook();
+
+    expect(result.current).toMatchInlineSnapshot(`
+      Object {
+        "high": 1,
+        "low": 0,
+        "max": 1,
+        "min": 0,
+        "optimum": 0.5,
+        "percent": 0,
+        "status": "safe",
+        "value": 0,
+      }
+    `);
+  });
+
+  it("meter state with value", function () {
+    const result = renderMeterStateHook({ value: 0.5 });
+
+    expect(result.current).toMatchInlineSnapshot(`
+      Object {
+        "high": 1,
+        "low": 0,
+        "max": 1,
+        "min": 0,
+        "optimum": 0.5,
+        "percent": 50,
+        "status": "safe",
+        "value": 0.5,
+      }
+    `);
+  });
+
+  it("meter state with custom props", function () {
+    const result = renderMeterStateHook({
+      min: 0,
+      low: 2.5,
+      value: 5,
+      high: 7.5,
+      max: 10,
+    });
+
+    expect(result.current).toMatchInlineSnapshot(`
+      Object {
+        "high": 7.5,
+        "low": 2.5,
+        "max": 10,
+        "min": 0,
+        "optimum": 5,
+        "percent": 50,
+        "status": "safe",
+        "value": 5,
+      }
+    `);
+  });
```

Comment:
Don't need to use this many snapshot testing, snapsshot testing only useful for very large objects where items are nested and have nested Array, Objects or Functions. 
And snapshots are useful for HTML markup testing.


Use plain objects instead :-

```js

  it("meter state with custom props", function () {
    const result = renderMeterStateHook({
      min: 0,
      low: 2.5,
      value: 5,
      high: 7.5,
      max: 10,
    });

    expect(result.current).toMatchObject({
      high: 7.5,
      low: 2.5,
      max: 10,
      min: 0,
      optimum: 5,
      percent: 50,
      status: "safe",
      value: 5,
    });
  });
```